sycl/source/detail/device_info.hpp

```diff
@@ -176,6 +184,30 @@ struct get_device_info<vector_class<info::fp_config>, param> {
   }
 };
 
+// Specialization for OpenCL version, splits the string returned by OpenCL
+template <> struct get_device_info<string_class, info::device::version> {
+  static string_class get(RT::PiDevice dev, const plugin &Plugin) {
+    string_class result =
+        get_device_info_string<info::device::version>::get(dev, Plugin);
+
+    // Extract OpenCL version from the returned string.
+    // For example, for the string "OpenCL 2.1 (Build 0)"
+    // return '2.1'.
+    auto dotPos = result.find('.');
+    if (dotPos == std::string::npos)
+      return result;
+
+    auto leftPos = result.rfind(' ', dotPos);
+    if (leftPos == std::string::npos)
+      leftPos = 0;
+    else
+      leftPos++;
+
+    auto rightPos = result.find(' ', dotPos);
+    return result.substr(leftPos, rightPos - leftPos);
```

Comment:
the function is not robust. If version string contains "bla X.Y.Z bla" it will report X.Y.Z which also violates spec.

Reply:
Yes, I know. It will also violate the spec if the plugin call returns the string without a dot, like 'XYZ'. What we should do in this case? Throwing an exception seems too strict, isn't it?

Reply:
can we make sure that if format of the version returned by BE does not fit our requirements (e.g. no dot, X.Y.Z, X.Y at the end of string) the user friendly string will be displayed (e.g. the whole string returned by BE, X.Y(.Z is ignored), X.Y).

Reply:
Ok. This is what I'm doing if the string doesn't have the dot. E.g. For string "bla XdotY bla" I return "bla XdotY bla".
I'm not sure how you define "user friendly string".